javascript/javascript3/week2/js3w2/js3w2_homework/js3w2-hw3_rewriteTime.js

```diff
@@ -0,0 +1,89 @@
+
+//Question3:
+// Rewrite setTimeout and navigator.geolocation.getCurrentPosition to promises.
+// The getCurrentPosition function is probably going to throw an error, 
+// but that is fine. As long as you can use it as a promise.
+
+
+// set Timeout Promise
+function setTimeoutPromise(time) {
+  return new Promise((resolve, reject) => {
+
+    if (isNaN(time)) {
+      reject('Rejected set Timeout promise ')
+    } else {
+    setTimeout(() => {
+      resolve('Resolved set Timeout promise')
+      }, time)
+    }
+  })
+}
+
+setTimeoutPromise('3000').then((resolveData) => {
+  console.log( resolveData);
+}).catch((rejectData) => {
+  console.log(rejectData);
+}); 
+
+
+// getCurrentLocation promise : TRY1
+function getCurrentLocation(resolveAfter) {
+  return new Promise((resolve, reject) => {
+    navigator.geolocation.getCurrentPosition(success, error,options);
+
+    var options = {
+      enableHighAccuracy: true,
+      timeout: 5000,
+      maximumAge: 0
+    };
+
+    let positionString;
+    function success(pos) {
+      const crd = pos.coords;
+      positionString = `Succes logged out after  ${resolveAfter} sec.
```

Comment:
instead of global variable you can return the value here

Reply:
ok